ee/clickhouse/queries/cohort_query.py

```diff
@@ -204,7 +205,16 @@ def _build_sources(self, subq: List[Tuple[str, str]]) -> Tuple[str, str]:
                 q += f"({subq_query}) {subq_alias}"
                 fields = f"{subq_alias}.person_id"
             elif prev_alias:  # can't join without a previous alias
-                q = f"{q} {full_outer_join_query(subq_query, subq_alias, f'{subq_alias}.person_id', f'{prev_alias}.person_id')}"
+                # TODO: decide when to use inner join
+                if (
+                    subq_alias == self.PERSON_TABLE_ALIAS
+                    and "person" not in [prop.type for prop in getattr(self._outer_property_groups, "flat", [])]
+                    and "cohort" not in [prop.type for prop in getattr(self._outer_property_groups, "flat", [])]
+                ):
+                    q = f"{q} {inner_join_query(subq_query, subq_alias, f'{subq_alias}.person_id', f'{prev_alias}.person_id')}"
+                else:
+                    q = f"{q} {full_outer_join_query(subq_query, subq_alias, f'{subq_alias}.person_id', f'{prev_alias}.person_id')}"
+
                 fields = if_condition(
                     f"{prev_alias}.person_id = '00000000-0000-0000-0000-000000000000'",
```

Comment:
Unrelated to this PR, but why do we have this `person_id` check?

Reply:
When you do a full outer join and there's no match on the join condition, one of the fields will be null'd by default so you need to optimistically look for the id across the field from both tables